test/SlowTests/Issues/RavenDB-22523.cs

```diff
@@ -0,0 +1,276 @@
+
+                // randomly decide whether to add silver w/gold tag
+                if (hasGold && random.Next(2) == 0)
+                {
+                    data.Tags.Add(silverTag);
+
+                    // Record how many documents have both the gold and silver tags
+                    excludeAllCounts++;
+                }
+
+                bulkInsert.Store(data);
+            }
+        }
+
+        // Prevent stale indexes
+        Indexes.WaitForIndexing(store);
+
+        // Filter out "exclude all" docs that have both the gold and silver tags
+        var filterValues = new[] { goldTag.Id, silverTag.Id };
+
+        // Query Corax Index
+        var coraxResults = GetResults<TestDocumentCoraxIndex>();
+
+        // Query Lucene Index
+        var luceneResults = GetResults<TestDocumentLuceneIndex>();
+
+        // Calculate Expected Total Results
+        var expectedTotalResults = numberOfDocsToCreate - excludeAllCounts;
+
+        Assert.Equal(take == int.MaxValue ? expectedTotalResults : take, luceneResults.ActualResultsCount);
+
+        Assert.Equal(take == int.MaxValue ? expectedTotalResults : take, coraxResults.ActualResultsCount);
+
+        // Validate Lucene Index Statistics.TotalResults
+        Assert.Equal(expectedTotalResults, luceneResults.TotalResults); // Success
+
+        // Validate Corax Index Statistics.TotalResults
+        Assert.Equal(expectedTotalResults, coraxResults.TotalResults); // Fails (seems to be returning the Collection size and not the excluded size)
+
+        (long ActualResultsCount, long TotalResults) GetResults<T>() where T : AbstractIndexCreationTask, new()
+        {
+            using var session = store.OpenSession();
+            var query = session.Advanced
+                .DocumentQuery<TestDocument, T>()
+                .Include(x => x.TagIds);
+
+            // Excludes All Query
+            query = query.AndAlso();
+            query = query.OpenSubclause();
+            query = query.Not.ContainsAll("TagIds", filterValues);
+            query = query.OrElse().Not.WhereExists("TagIds");
+            query = query.CloseSubclause();
+            query = query.Skip(0).Take(take);
+            query = query.Statistics(out var stats);
+            
+            var results = query.OfType<TestDocument>().ToList();
+            return (results.Count, stats.TotalResults);
+        }
+    }
+    
+    /**
+     * Issue with Corax Index and IDocumentQuery - Excludes Any.
+     *
+     * This test is designed to show an issue with the Statistics TotalResults value
+     * when using an Excludes Any IDocumentQuery. The expected value should be smaller than
+     * the total number of documents in the Collection as well as accurate to the filtered value(s).
+     */
+    [RavenTheory(RavenTestCategory.Indexes | RavenTestCategory.Corax)]
+    [InlineData(5000, 10)]
+    [InlineData(5000, int.MaxValue)]
```

Comment:
Did you verify it with 2000000 docs as well?

https://issues.hibernatingrhinos.com/issue/RavenDB-22523/Corax-Too-many-results-returned-for-queries-doing-exclusions#focus=Comments-67-1053525.0-0

Reply:
![image](https://github.com/ravendb/ravendb/assets/86351904/d47b0275-b471-4235-ad76-b86f057df007)